src/OpenTelemetry.AutoInstrumentation/Instrumentations/GraphQL/GraphQLCommon.cs

```diff
@@ -45,42 +40,23 @@ internal class GraphQLCommon
 
     private static readonly ILogger Log = OtelLogging.GetLogger();
 
-    internal static Activity CreateActivityFromValidate(IDocument document)
-    {
-        Activity activity = null;
-
-        try
-        {
-            var tags = new GraphQLTags();
-            activity = ActivitySource.StartActivityWithTags(ValidateOperationName, tags);
-
-            tags.Source = document.OriginalQuery;
-        }
-        catch (Exception ex)
-        {
-            Log.Error(ex, "Error creating or populating scope.");
-        }
-
-        return activity;
-    }
-
     internal static Activity CreateActivityFromExecuteAsync(IExecutionContext executionContext)
     {
         Activity activity = null;
 
         try
         {
-            string source = executionContext.Document.OriginalQuery;
+            string query = executionContext.Document.OriginalQuery;
             string operationName = executionContext.Operation.Name;
-            string operationType = executionContext.Operation.OperationType.ToString();
+            string operationType = executionContext.Operation.OperationType.ToString().ToLowerInvariant();
+            string operation = GetOperation(operationName, operationType);
 
             var tags = new GraphQLTags();
-            activity = ActivitySource.StartActivityWithTags(ExecuteOperationName, tags);
-            activity.SetResourceName($"{operationType} {operationName ?? "operation"}");
-
-            tags.Source = source;
+            tags.Document = query; // TODO: Sanitization is recommended.
```

Comment:
We should give an abilty to explicitly enable/disable the addition of the `graphql.document` attribute by using an env var. IMO it should be disabled by default to avoid processing PII. It is fine for me if we just create an issue for it, but I would love to have it addressed in a timely manner 😉

Reply:
I would still keep it true by default, just because the spec says sanitization is optional (`may be`). So according to the wording to me it seems that PII processing is not so big deal **for default user**.